diff --git a/dist/index.mjs b/dist/index.mjs
index eb3a02f919c190961256b74d9a927e6972666561..f86faf070740e88e439cac0c4168308524c2554e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -410,7 +410,10 @@ function loadWasm(options) {
       return instance;
     });
   }
-  initPromise = _load();
+  initPromise = _load().catch((error) => {
+    initPromise = undefined;
+    throw error;
+  });
   return initPromise;
 }
 function _makeArrayBufferLoader(data) {
